### ISSUE TITLE
crowdstrike_detection table - update required due to API deprecation

### ISSUE THREAD

**Issue by l-teles** [turbot/steampipe-plugin-crowdstrike#49]
**Describe the bug**
CrowdStrike has deprecated the Detections API, which requires the `crowdstrike_detection` table to be populated differently. 

- `GetDetectSummaries` was replaced by `alerts.QueryV2`

- `GetAggregateDetects` was replaced by `alerts.GetAggregateV2`

Please check [this](https://github.com/CrowdStrike/gofalcon/issues/453) for more details. 

**Plugin version (`steampipe plugin list`)**
v1.0.0

**Comment by misraved** [comment]
Thanks @l-teles for raising this issue and apologies for the delayed response!!

I will take a look at the new SDK version and give a rough estimate of when we will be able to make the changes (need to verify if there are no breaking changes).

Thanks!!

**Comment by ParthaI** [comment]
Hello @l-teles, have you had a chance to test this on the PR branch yet?

Your feedback would be much appreciated—thanks!

**Comment by ffalor** [comment]
Hey, I added a document that covers these endpoints to this comment: https://github.com/CrowdStrike/gofalcon/issues/453#issuecomment-3192212883

**Comment by ParthaI** [comment]
Hello @ffalor, thank you so much for sharing the docs. I was reviewing them and noted a few points:

* The **Detection API** has been deprecated. We now need to use the **Alert API**.
* The document provides the following guidance:

  1. **Get a list of IDs for EPP alerts**

     ```bash
     curl -X GET "https://api.crowdstrike.com/alerts/queries/alerts/v2?filter=product:'epp'&limit=5" \
       -H 'Authorization: bearer eyJhbGci...xYg1NNI' \
       -H 'Accept: application/json'
     ```
  2. **Retrieve detailed information for a specific alert ID**

     ```bash
     curl -X POST "https://api.crowdstrike.com/alerts/entities/alerts/v2" \
       -H 'Authorization: bearer eyJhbGci...xYg1NNI' \
       -H 'Accept: application/json' \
       -H 'Content-Type: application/json' \
       -d '{
         "composite_ids": [
           "d615xxxxxxxx2158:ind:9a8dxxxxxxxxc74c:1336xxxxxxxx1294-32-7878xxxxxxxx1122"
         ]
       }'
     ```

For the first curl command, the filter value is set to `product:'epp'`, which I understand will only return **EPP product alerts**.

My question: **What filter value should I use to return *all available alerts*?**

According to the SDK comment:

```
/* Filter.

	     Filter Alerts using a query in Falcon Query Language (FQL).Filter fields can be any keyword field that is part of #domain.Alert
	An asterisk wildcard `*` includes all results.
	Empty value means to not filter on anything.
	Most commonly used filter fields that supports exact match: cid, id, aggregate_id, product, type, pattern_id, platform ...
	Most commonly used filter fields that supports wildcard (*): assigned_to_name, assigned_to_uuid, tactic_id, technique ...
	Most commonly filter fields that supports range comparisons (>, <, >=, <=): severity, created_timestamp, timestamp, updated_timestamp...
	All filter fields and operations support negation (!).


	The full list of valid filter options is extensive. Review it in our [documentation inside the Falcon console](https://falcon.crowdstrike.com/documentation/45/falcon-query-language-fql).
	*/
```

Should we use `?filter=*`, or just omit the filter entirely to get all alerts?

Also, it would be greatly appreciated if you could share the [Falcon Query Language documentation](https://falcon.crowdstrike.com/documentation/45/falcon-query-language-fql).


Thanks again!

**Comment by ffalor** [comment]
I would omit it, have you seen the page about our FQL from [falconpy](https://www.falconpy.io/Usage/Falcon-Query-Language.html#using-fql-in-a-filter)?

It does a good job explaining FQL and how to use it. The python examples are still relevant for gofalcon since both falconpy and goFalcon pass the filter directly to the API. It's pretty much verbatim our internal doc with some real world examples.

**Comment by ParthaI** [comment]
Yup @ffalor, I could see the doc you shared above. I will take a look and try to make the changes. Thanks again for your help!
